packages/tasit-sdk/src/Decentraland.test.js

```diff
@@ -142,109 +94,493 @@ describe("Decentraland", () => {
+          const spender = MARKETPLACE_ADDRESS;
+          const allowance = await mana.allowance(owner, spender);
+          expect(`${allowance}`).to.equal(`${manaAmountForShopping}`);
+        });
+
+        it("should buy an estate", async () => {
+          const {
+            assetId,
+            nftAddress,
+            seller,
+            priceInWei,
+            expiresAt,
+          } = estateForSale;
+
+          const buyerAddress = GNOSIS_SAFE_ADDRESS;
+          await checkAsset(estate, mana, estateForSale, buyerAddress);
+
+          await confirmBalances(estate, [GNOSIS_SAFE_ADDRESS], [0]);
+
+          // Gnosis Safe should approve Marketplace to spend its MANA Tokens
+          // TODO: ephemeralWallet should broadcast this action
+          const contractAddress = marketplace.getAddress();
+          const contractABI = marketplace.getABI();
+          const functionName = "safeExecuteOrder";
+          const fingerprint = await estate.getFingerprint(`${assetId}`);
+          const argsArray = [
+            nftAddress,
+            `${assetId}`,
+            `${priceInWei}`,
+            `${fingerprint}`,
+          ];
+
+          gnosisSafe.setSigners([gnosisSafeOwner]);
+          gnosisSafe.setWallet(gnosisSafeOwner);
+          const executeOrderAction = gnosisSafe.customContractAction(
+            contractAddress,
+            contractABI,
+            functionName,
+            argsArray
+          );
+
+          const onFailed = message => {
+            const { data } = message;
+            const { args } = data;
+            const { txHash } = args;
+          };
+          gnosisSafe.once("ExecutionFailed", onFailed);
+
+          await executeOrderAction.waitForNonceToUpdate();
+
+          await mineBlocks(provider, 1);
+
+          await confirmBalances(estate, [GNOSIS_SAFE_ADDRESS], [1]);
+        });
+
+        it("should buy a parcel of land", async () => {
+          const {
+            assetId,
+            nftAddress,
+            seller,
+            priceInWei,
+            expiresAt,
+          } = landForSale;
+
+          checkAsset(land, mana, landForSale, GNOSIS_SAFE_ADDRESS);
+
+          await confirmBalances(land, [GNOSIS_SAFE_ADDRESS], [0]);
+
+          // Gnosis Safe should approve Marketplace to spend its MANA Tokens
+          // TODO: ephemeralWallet should broadcast this action
+          const contractAddress = marketplace.getAddress();
+          const contractABI = marketplace.getABI();
+          const functionName = "safeExecuteOrder";
+          // LANDRegistry contract doesn't implement getFingerprint function
+          const fingerprint = "0x";
+          const argsArray = [
+            nftAddress,
+            `${assetId}`,
+            `${priceInWei}`,
+            `${fingerprint}`,
+          ];
+          const ethersAmount = "0";
+          gnosisSafe.setSigners([gnosisSafeOwner]);
+          gnosisSafe.setWallet(gnosisSafeOwner);
```

Comment:
There's so much boilerplate for each transaction. I wonder if a convenience function is warranted.

Reply:
Yes. I think that will be better when we have some thing like that:
```javascript
// draft
const marketplaceAction = marketplace.safeExecuteOrder(...);
const marketplaceActionData = marketplaceAction.toData() // or .getData() or .data
const gnosisSafeAction = gnosisSafe.prepare(actionData);
gnosisSafeAction.send();

```

Reply:
Yeah, that looks good!